profitbricks-rest/src/test/java/org/apache/jclouds/profitbricks/rest/features/NicApiLiveTest.java

```diff
@@ -0,0 +1,153 @@
+  
+   @BeforeClass
+   public void setupTest() {
+      dataCenter = createDataCenter();
+      assertDataCenterAvailable(dataCenter);
+            
+      testServer = api.serverApi().createServer(
+         Server.Request.creatingBuilder()
+            .dataCenterId(dataCenter.id())
+            .name("jclouds-node")
+            .cores(1)
+            .ram(1024)
+            .build());
+      
+      assertNodeAvailable(ServerRef.create(dataCenter.id(), testServer.id()));
+   }
+   
+   @AfterClass(alwaysRun = true)
+   public void teardownTest() {
+      if (dataCenter != null)
+         deleteDataCenter(dataCenter.id());
+   }
+     
+   @Test
+   public void testCreateNic() {
+      assertNotNull(dataCenter);
+            
+      testNic = nicApi().createNic(
+              Nic.Request.creatingBuilder()
+              .dataCenterId(dataCenter.id())
+              .serverId(testServer.id())
+              .name("jclouds-nic")
+              .lan(1)
+              .build());
+
+      assertNotNull(testNic);
+      assertEquals(testNic.properties().name(), "jclouds-nic");
+      assertNicAvailable(testNic);
+   }
+   
+
+   @Test(dependsOnMethods = "testCreateNic")
+   public void testGetNic() {
+      Nic nic = nicApi().getNic(dataCenter.id(), testServer.id(), testNic.id());
+
+      assertNotNull(nic);
+      assertEquals(nic.id(), testNic.id());
+   }
+
+   @Test(dependsOnMethods = "testCreateNic")
+   public void testList() {
+      List<Nic> nics = nicApi().getList(dataCenter.id(), testServer.id());
+
+      assertNotNull(nics);
+      assertFalse(nics.isEmpty());
+      assertEquals(nics.size(), 1);
+   }
+   
+   @Test(dependsOnMethods = "testGetNic")
+   public void testUpdateNic() {
+      assertDataCenterAvailable(dataCenter);
+      
+      nicApi().updateNic(
+              Nic.Request.updatingBuilder()
+              .dataCenterId(testNic.dataCenterId())
+              .serverId(testServer.id())
+              .id(testNic.id())
+              .name("apache-nic")
+              .build());
+
+      assertNicAvailable(testNic);
+      
+      Nic nic = nicApi().getNic(dataCenter.id(), testServer.id(), testNic.id());
+      
+      assertEquals(nic.properties().name(), "apache-nic");
+   }
+   
+
+   @Test(dependsOnMethods = "testUpdateNic")
+   public void testDeleteNic() {
+      nicApi().deleteNic(testNic.dataCenterId(), testServer.id(), testNic.id());
+      assertNicRemoved(testNic);
+   } 
+   
+   private void assertNicAvailable(Nic nic) {
+      assertPredicate(new Predicate<String>() {
+         @Override
+         public boolean apply(String args) {
+            String[] params = args.split(",");
+            Nic nic = nicApi().getNic(params[0], params[1], params[2]);
+            
+            if (nic == null || nic.metadata() == null)
+               return false;
+            
+            return nic.metadata().state() == State.AVAILABLE;
+         }
+      }, complexId(nic.dataCenterId(), testServer.id(), nic.id()));
```

Comment:
Why assembling this into a String to just split it immediately? Why not passing directly the Nic object to the predicate?